paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java

```diff
@@ -3016,17 +3018,6 @@ public void updateCommands() {
         this.getHandle().server.getCommands().sendCommands(this.getHandle());
     }
 
-    @Override
-    public void openBook(ItemStack book) {
-        Preconditions.checkArgument(book != null, "ItemStack cannot be null");
-        Preconditions.checkArgument(book.getType() == Material.WRITTEN_BOOK, "ItemStack Material (%s) must be Material.WRITTEN_BOOK", book.getType());
-
-        ItemStack hand = this.getInventory().getItemInMainHand();
-        this.getInventory().setItemInMainHand(book);
-        this.getHandle().openItemGui(org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(book), net.minecraft.world.InteractionHand.MAIN_HAND);
-        this.getInventory().setItemInMainHand(hand);
-    }
```

Comment:
I would not override this. The behavior is a bit different here in that it resolves the book item.
Your improvement to openBook on adventure is fine tho, but I would keep this the same.

Calling openItemGui is a bit different, so this should be kept the same, only adventure book should be changed.

Reply:
ok then rollback a use the same behaviour for "item" and "adventure book creating a item"